### PR TITLE
Set correct C flags when compiling with GCC and C++ not enabled

### PR DESCRIPTION
**Pull request by andrewhop**
### Issues:
Resolves CryptoAlg-716

### Description of changes: 
We added support to build just libcrypto and skip building libssl and all the testing. In that mode we don't enable CXX so the `CMAKE_COMPILER_IS_GNUCXX` variable is never true and we did not set all the available flags for GCC. This change checks the `CMAKE_C_COMPILER_ID` variable like we do for Clang and use that to make all the decisions for which flags to set.

### Testing:
Tested locally with gcc 7 and clang 9. Tried the combination of both compilers with and without libssl/testing. Also tested with gcc 4 in the Centos image:
```
desktop git:(compiler-fix)✗ > docker run -v `pwd`:`pwd` -w `pwd` -it centos-7:gcc-4x
[root@e9808e73e133 aws-lc]# cd build/
[root@e9808e73e133 build]# cmake -GNinja -DBUILD_TESTING=OFF -DBUILD_LIBSSL=OFF ../
-- The C compiler identification is GNU 4.8.5
-- Check for working C compiler using: Ninja
-- Check for working C compiler using: Ninja -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Found Perl: /usr/bin/perl (found version "5.16.3")
pkgconfig not found. Disabling unwind tests.
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ANT.AMAZON.COM/andhop/workplace/oss/aws-lc/build
[root@e9808e73e133 build]# ninja
[1/1] Re-running CMake...
pkgconfig not found. Disabling unwind tests.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ANT.AMAZON.COM/andhop/workplace/oss/aws-lc/build
[291/291] Linking C static library decrepit/libdecrepit.a
[root@e9808e73e133 build]# 
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -151,17 +157,14 @@ elseif((BORINGSSL_PREFIX AND BORINGSSL_PREFIX_SYMBOLS) AND NOT GO_EXECUTABLE)
...
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  set(CLANG 1)
-endif()
-
 if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
   set(EMSCRIPTEN 1)
 endif()
 
-if(CMAKE_COMPILER_IS_GNUCXX OR CLANG)
+if(GCC OR CLANG)
```
Comment (justinboswell) [inline CMakeLists.txt:164]: AHA, nice!

**Review by justinboswell — DISMISSED** [review]

**Comment by bryce-shang** [comment]
> Tested locally with gcc 7 and clang 9. Tried the combination of both compilers with and without libssl/testing.

CryptoAlg-716 is found when testing with gcc4. Testing with other compilers didn't report the c99 error.

File: CMakeLists.txt [hunk 2]
```diff
@@ -792,3 +787,7 @@ configure_file("cmake/awslc-config.cmake"
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/awslc-config.cmake"
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/"
     COMPONENT Development)
+
+if(NOT MSVC AND NOT CLANG AND NOT GCC)
+  message(STATUS "Alternative compiler '${CMAKE_C_COMPILER_ID}' detected. Not all flags may be set, check final options with 'cmake --build . -- VERBOSE=1'")
+endif()
```
Comment (andrew-kaufman) [inline CMakeLists.txt:793]: Newline at end of file?

**Review by andrew-kaufman — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -151,17 +157,14 @@ elseif((BORINGSSL_PREFIX AND BORINGSSL_PREFIX_SYMBOLS) AND NOT GO_EXECUTABLE)
...
-
 if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
   set(EMSCRIPTEN 1)
 endif()
 
-if(CMAKE_COMPILER_IS_GNUCXX OR CLANG)
+if(GCC OR CLANG)
   # Note clang-cl is odd and sets both CLANG and MSVC. We base our configuration
   # primarily on our normal Clang one.
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
```
Comment (bryce-shang) [inline CMakeLists.txt:167]: Does awslc not support `c11`?

```
- set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c11")
```

**Review by bryce-shang — APPROVED** [review]

**Pull request merged**